include/boost/multiprecision/rational_adaptor.hpp

```diff
@@ -247,106 +323,780 @@ inline typename std::enable_if<number_category<R>::value == number_kind_floating
+
+   if (eval_get_sign(a.num()) == 0)
+   {
+      BOOST_THROW_EXCEPTION(std::overflow_error("Integer division by zero"));
+      return;
+   }
+   if (&a == &result)
+   {
+      eval_multiply_imp(result.denom(), result.num(), b);
+      result.num().swap(result.denom());
+   }
+   else
+      eval_multiply_imp(result, a.denom(), a.num(), b);
+
+   if (eval_get_sign(result.denom()) < 0)
+   {
+      result.num().negate();
+      result.denom().negate();
+   }
+}
+
+template <class Backend, class Arithmetic>
+typename std::enable_if<std::is_convertible<Arithmetic, Backend>::value&& std::is_integral<Arithmetic>::value>::type
+eval_divide(rational_adaptor<Backend>& result, Arithmetic arg)
+{
+   if (arg == 0)
+   {
+      BOOST_THROW_EXCEPTION(std::overflow_error("Integer division by zero"));
+      return;
+   }
+   else if (arg == 1)
+      return;
+   else if (is_minus_one(arg))
+   {
+      result.negate();
+      return;
+   }
+   if (eval_get_sign(result) == 0)
+   {
+      return;
+   }
+
+
+   using default_ops::eval_multiply;
+   using default_ops::eval_gcd;
+   using default_ops::eval_convert_to;
+   using default_ops::eval_divide;
+
+   Backend gcd, t;
+   Arithmetic integer_gcd;
+   eval_gcd(gcd, result.num(), arg);
+   eval_convert_to(&integer_gcd, gcd);
+   arg /= integer_gcd;
+
+   eval_multiply(t, result.denom(), boost::multiprecision::detail::unsigned_abs(arg));
+   result.denom() = std::move(t);
+   if (arg < 0)
+   {
+      result.num().negate();
+   }
+   if (integer_gcd > 1)
+   {
+      eval_divide(t, result.num(), integer_gcd);
+      result.num() = std::move(t);
+   }
+}
+
+template <class Backend, class Arithmetic>
+typename std::enable_if<std::is_convertible<Arithmetic, Backend>::value&& std::is_integral<Arithmetic>::value>::type
+   eval_divide(rational_adaptor<Backend>& result, const rational_adaptor<Backend>& a, Arithmetic arg)
+{
+   if (&result == &a)
+      return eval_divide(result, arg);
+   if (arg == 0)
+   {
+      BOOST_THROW_EXCEPTION(std::overflow_error("Integer division by zero"));
+      return;
+   }
+   else if (arg == 1)
+   {
+      result = a;
+      return;
+   }
+   else if (is_minus_one(arg))
+   {
+      result = a;
+      result.num().negate();
+      return;
+   }
+
+   if (eval_get_sign(a) == 0)
+   {
+      result = a;
+      return;
+   }
+
+   using default_ops::eval_multiply;
+   using default_ops::eval_divide;
+   using default_ops::eval_gcd;
+   using default_ops::eval_convert_to;
+
+   Backend gcd;
+   Arithmetic integer_gcd;
+   eval_gcd(gcd, a.num(), arg);
+   eval_convert_to(&integer_gcd, gcd);
+   arg /= integer_gcd;
+   eval_multiply(result.denom(), a.denom(), boost::multiprecision::detail::unsigned_abs(arg));
+
+   if (integer_gcd > 1)
+   {
+      eval_divide(result.num(), a.num(), integer_gcd);
+   }
+   else
+      result.num() = a.num();
+   if (arg < 0)
+   {
+      result.num().negate();
+   }
+}
+
+//
+// Multiply and add/subtract as one:
+//
+#if 0
```

Comment:
Don't wanna just remove these?

Reply:
I'm conflicted, there is a simplified formula for multiply&add of rationals, so I might yet implement these....